File: TeamCode/src/main/java/org/firstinspires/ftc/teamcode/Components/Intake.java
package org.firstinspires.ftc.teamcode.Components;

import com.qualcomm.robotcore.hardware.CRServo;
import com.qualcomm.robotcore.hardware.ColorRangeSensor;
import com.qualcomm.robotcore.hardware.DcMotor;
import com.qualcomm.robotcore.hardware.DcMotorSimple;
import com.qualcomm.robotcore.hardware.HardwareMap;
import com.qualcomm.robotcore.hardware.Servo;

import org.firstinspires.ftc.robotcore.external.navigation.DistanceUnit;

import static org.firstinspires.ftc.teamcode.Components.Details.opModeType;

public class Intake implements Component {
    public enum IntakeState {
        SAFE,
        CUSTOM_VALUE,
        IDLE,
    }
    IntakeState intakeState = IntakeState.CUSTOM_VALUE;
    DcMotor intakeMotor;
    CRServo intakeL, intakeR;
    Servo drop;


    public Intake(HardwareMap hardwareMap) {
        intakeMotor = hardwareMap.get(DcMotor.class, "intake");
        intakeMotor.setDirection(DcMotorSimple.Direction.REVERSE);
        intakeMotor.setMode(DcMotor.RunMode.STOP_AND_RESET_ENCODER);
        intakeMotor.setMode(DcMotor.RunMode.RUN_WITHOUT_ENCODER);
        intakeL = hardwareMap.crservo.get("intakeL");
        intakeR = hardwareMap.crservo.get("intakeR");
        drop = hardwareMap.servo.get("drop");
        intakeR.setDirection(DcMotorSimple.Direction.REVERSE);

    }

    @Override
    public void update() {
    }

    public void setIntakeState(IntakeState intakeState) {
        this.intakeState = intakeState;
    }

    public IntakeState getIntakeState() {
        return intakeState;
    }

    private void setVectors(double power) {
        intakeL.setPower(power);
        intakeR.setPower(power);
    }

    //use these two functions so that it drops intake and then raises the dropper
    public void dropIntake(){ drop.setPosition(0.6);}
    public void raiseIntake(){drop.setPosition(1);}

    //public double getDropPosition(){return drop.getPosition();}
    public void setPower(double power) {
        setIntakeState(IntakeState.CUSTOM_VALUE);
        intakeMotor.setPower(power);
        setVectors(power);
    }
}

File: TeamCode/src/main/java/org/firstinspires/ftc/teamcode/Components/Intake.kt
package org.firstinspires.ftc.teamcode.Components

import com.qualcomm.robotcore.hardware.*

class Intake(hardwareMap: HardwareMap) : Component {
    enum class IntakeState {
        SAFE, CUSTOM_VALUE, IDLE
    }

    var intakeState = IntakeState.CUSTOM_VALUE
    var intakeMotor: DcMotor
    var intakeL: CRServo
    var intakeR: CRServo
    var drop: Servo
    override fun update() {}
    private fun setVectors(power: Double) {
        intakeL.power = power
        intakeR.power = power
    }

    //use these two functions so that it drops intake and then raises the dropper
    fun dropIntake() {
        drop.position = 0.6
    }

    fun raiseIntake() {
        drop.position = 1.0
    }

    //public double getDropPosition(){return drop.getPosition();}
    fun setPower(power: Double) {
        intakeState = IntakeState.CUSTOM_VALUE
        intakeMotor.power = power
        setVectors(power)
    }

    init {
        intakeMotor = hardwareMap.get(DcMotor::class.java, "intake")
        intakeMotor.direction = DcMotorSimple.Direction.REVERSE
        intakeMotor.mode = DcMotor.RunMode.STOP_AND_RESET_ENCODER
        intakeMotor.mode = DcMotor.RunMode.RUN_WITHOUT_ENCODER
        intakeL = hardwareMap.crservo["intakeL"]
        intakeR = hardwareMap.crservo["intakeR"]
        drop = hardwareMap.servo["drop"]
        intakeR.direction = DcMotorSimple.Direction.REVERSE
    }
}